server/blob/blob-cassandra/src/main/java/org/apache/james/blob/cassandra/CassandraDumbBlobStore.java

```diff
@@ -0,0 +1,223 @@
+import org.apache.james.blob.api.ObjectStoreIOException;
+import org.apache.james.blob.cassandra.utils.DataChunker;
+import org.apache.james.util.ReactorUtils;
+
+import com.github.fge.lambdas.Throwing;
+import com.google.common.base.Preconditions;
+import com.google.common.io.ByteSource;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+
+public class CassandraDumbBlobStore implements DumbBlobStore {
+
+    private static final int PREFETCH = 16;
+    private static final int MAX_CONCURRENCY = 1;
+    private final CassandraDefaultBucketDAO defaultBucketDAO;
+    private final CassandraBucketDAO bucketDAO;
+    private final DataChunker dataChunker;
+    private final CassandraConfiguration configuration;
+    private final HashBlobId.Factory blobIdFactory;
+    private final BucketName defaultBucket;
+
+    @Inject
+    CassandraDumbBlobStore(CassandraDefaultBucketDAO defaultBucketDAO,
+                           CassandraBucketDAO bucketDAO,
+                           CassandraConfiguration cassandraConfiguration,
+                           HashBlobId.Factory blobIdFactory,
+                           BucketName defaultBucket) {
+        this.defaultBucketDAO = defaultBucketDAO;
+        this.bucketDAO = bucketDAO;
+        this.configuration = cassandraConfiguration;
+        this.blobIdFactory = blobIdFactory;
+        this.defaultBucket = defaultBucket;
+        this.dataChunker = new DataChunker();
+    }
+
+    @Override
+    public InputStream read(BucketName bucketName, BlobId blobId) throws ObjectStoreIOException, ObjectNotFoundException {
+        return ReactorUtils.toInputStream(readBlobParts(bucketName, blobId));
+    }
+
+    @Override
+    public Mono<byte[]> readBytes(BucketName bucketName, BlobId blobId) {
+        return readBlobParts(bucketName, blobId)
+            .collectList()
+            .map(this::byteBuffersToBytesArray);
+    }
+
+    @Override
+    public Mono<Void> save(BucketName bucketName, BlobId blobId, byte[] data) {
+        Preconditions.checkNotNull(data);
+
+        return saveAsMono(bucketName, blobId, data);
+    }
+
+    private Mono<Void> saveAsMono(BucketName bucketName, BlobId blobId, byte[] data) {
+        return saveBlobParts(bucketName, data, blobId)
+            .flatMap(numberOfChunk -> saveBlobPartReference(bucketName, blobId, numberOfChunk));
+    }
+
+    private Mono<Integer> saveBlobParts(BucketName bucketName, byte[] data, BlobId blobId) {
+
+        Stream<Pair<Integer, ByteBuffer>> chunks = dataChunker.chunk(data, configuration.getBlobPartSize());
+        return Flux.fromStream(chunks)
+            .publishOn(Schedulers.elastic(), PREFETCH)
+            .flatMap(pair -> writePart(bucketName, blobId, pair.getKey(), pair.getValue())
+                .then(Mono.just(getChunkNum(pair))))
+            .collect(Collectors.maxBy(Comparator.comparingInt(x -> x)))
+            .flatMap(Mono::justOrEmpty)
```

Comment:
Use handle please!

Reply:
this code doesn't exist in later commit. Thanks for the hint thought.